pkg/resources/apiserver/deployment.go

```diff
@@ -569,10 +575,18 @@ func getVolumeMounts(isKonnectivityEnabled, isEncryptionEnabled bool) []corev1.V
 		})
 	}
 
+	if isAuditWebhookEnabled {
+		vms = append(vms, corev1.VolumeMount{
+			Name:      resources.AuditWebhookSecretName,
```

Comment:
Shouldn't this value come out of the Cluster spec? The reconciler is setting

```
c.Spec.AuditLogging.WebhookBackend.AuditWebhook = &corev1.SecretReference{
			Name:      resources.AuditWebhookSecretName,
			Namespace: c.Status.NamespaceName,
		}
```

so we should not dupllicate the Go constant here.

Reply:
We are not dulicating the go constant, they are from the same source in both the places.

Reply:
I don't quite understand what the value in the ClusterSpec is then good for. It seems that we're remembering explicitly what secret to use, per cluster, and then ignore the per-cluster value in this reconciler code, which always just uses the Go constant. Then we can just remove ClusterSpec.AuditLogging.WebhookBackend.AuditWebhook from the CRD.

Reply:
For cluster spec it's not very useful as the secret name will be same for all the cluster, in the dc spec it is more useful as admins can specify any secret if they enforce the webhook config but in both cluster and dc spec we are using the same struct so I am setting the same in the cluster spec as well. We can keep it this way or keep it in the crd but not set it in the cluster spec or remove it from the crd altogether, please let me know which one sounds the best to you.

Reply:
The field containing the webhook config secret has been made mandatory, it will get set based on the values provided by the admin.